core/src/snapshot_packager_service.rs

```diff
@@ -72,32 +78,55 @@ impl SnapshotPackagerService {
                         break;
                     }
 
-                    let snapshot_package = pending_snapshot_package.lock().unwrap().take();
-                    if snapshot_package.is_none() {
-                        std::thread::sleep(Duration::from_millis(100));
+                    let Some((
+                        snapshot_package,
+                        num_outstanding_snapshot_packages,
+                        num_re_enqueued_snapshot_packages,
+                    )) = Self::get_next_snapshot_package(&snapshot_package_sender, &snapshot_package_receiver) else {
+                        std::thread::sleep(Self::LOOP_LIMITER);
                         continue;
-                    }
-                    let snapshot_package = snapshot_package.unwrap();
-
-                    // Archiving the snapshot package is not allowed to fail.
-                    // AccountsBackgroundService calls `clean_accounts()` with a value for
-                    // last_full_snapshot_slot that requires this archive call to succeed.
-                    snapshot_utils::archive_snapshot_package(
-                        &snapshot_package,
-                        &snapshot_config.full_snapshot_archives_dir,
-                        &snapshot_config.incremental_snapshot_archives_dir,
-                        snapshot_config.maximum_full_snapshot_archives_to_retain,
-                        snapshot_config.maximum_incremental_snapshot_archives_to_retain,
-                    )
-                    .expect("failed to archive snapshot package");
-
-                    if let Some(snapshot_gossip_manager) = snapshot_gossip_manager.as_mut() {
-                        snapshot_gossip_manager.push_snapshot_hash(
-                            snapshot_package.snapshot_type,
-                            (snapshot_package.slot(), snapshot_package.hash().0),
-                        );
-                    }
+                    };
+                    info!("handling snapshot package: {snapshot_package:?}");
```

Comment:
How frequent is this? If memory serves, this is once every 400 slots or so, which would be once every 2-3 minutes. Probably okay to be INFO logging level, but might consider kicking it down to DEBUG

Reply:
Makes sense. We can always dial it back later